bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutation.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved. Licensed under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with the License. You may obtain
+ * a copy of the License at http://www.apache.org/licenses/LICENSE-2.0 Unless required by applicable
+ * law or agreed to in writing, software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ * for the specific language governing permissions and limitations under the License.
+ */
+package com.google.cloud.bigtable.grpc.async;
+
+import java.util.concurrent.ExecutionException;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import com.google.bigtable.v1.MutateRowRequest;
+import com.google.bigtable.v1.MutateRowsRequest;
+import com.google.bigtable.v1.MutateRowsRequest.Entry;
+import com.google.bigtable.v1.MutateRowsResponse;
+import com.google.bigtable.v1.Mutation;
+import com.google.bigtable.v1.Mutation.SetCell;
+import com.google.common.util.concurrent.SettableFuture;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Empty;
+import com.google.rpc.Status;
+import com.google.rpc.Status.Builder;
+
+/**
+ * Tests for {@link BulkMutation}
+ */
+@RunWith(JUnit4.class)
+public class TestBulkMutation {
```

Comment:
Like you mentioned, might be good to have a test that covers the unexpected entry count case.